packages/hurl/src/main.rs

```diff
@@ -48,7 +48,13 @@ struct HurlRun {
     content: String,
     /// Content's source file
     filename: Input,
-    hurl_result: HurlResult,
+    hurl_results: Vec<HurlResult>,
+}
+
+impl HurlRun {
+    pub fn last_hurl_result(&self) -> &HurlResult {
+        self.hurl_results.last().expect("at least one hurl result")
+    }
```

Comment:
Alternatively, have one final `hurl_result` and additional optional `hurl_result_attempts`.